x-pack/plugins/apm/public/components/app/Settings/agent_keys/index.tsx

```diff
@@ -0,0 +1,180 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import React, { Fragment } from 'react';
+import { isEmpty } from 'lodash';
+import { i18n } from '@kbn/i18n';
+import {
+  EuiText,
+  EuiSpacer,
+  EuiTitle,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiEmptyPrompt,
+  EuiButton,
+  EuiLoadingSpinner,
+} from '@elastic/eui';
+import { useFetcher, FETCH_STATUS } from '../../../../hooks/use_fetcher';
+import { PermissionDenied } from './prompts/permission_denied';
+import { ApiKeysNotEnabled } from './prompts/api_keys_not_enabled';
+import { AgentKeysTable } from './agent_keys_table';
+
+const INITIAL_DATA = {
+  areApiKeysEnabled: false,
+  isAdmin: false,
+  canManage: false,
+};
+
+export function AgentKeys() {
+  return (
+    <Fragment>
+      <EuiText color="subdued">
+        {i18n.translate('xpack.apm.settings.agentKeys.descriptionText', {
+          defaultMessage:
+            'View and delete agent keys. An agent key sends requests on behalf of a user.',
+        })}
+      </EuiText>
+      <EuiSpacer size="m" />
+      <EuiFlexGroup justifyContent="spaceBetween">
+        <EuiFlexItem grow={false}>
+          <EuiTitle>
+            <h2>
+              {i18n.translate('xpack.apm.settings.agentKeys.title', {
+                defaultMessage: 'Agent keys',
+              })}
+            </h2>
+          </EuiTitle>
+        </EuiFlexItem>
```

Comment:
There will be another item in the flex group that will wrap a `Create agent key` button.